File: src/main/java/org/kite9/diagram/model/style/BorderTraversal.java
package org.kite9.diagram.model.style;

public enum BorderTraversal {

	ALWAYS, LEAVING, NONE;
	
	public static BorderTraversal reduce(BorderTraversal a, BorderTraversal b) {
		if (a==null) {
			return b;
		} else if (b == null) {
			return a;
		} else if ((a != ALWAYS) && (b != ALWAYS)) {
			return NONE;
		} else {	
			return ALWAYS;
		}
	}
	
}

File: k9layout/src/main/kotlin/org/kite9/diagram/model/style/BorderTraversal.kt
package org.kite9.diagram.model.style

enum class BorderTraversal {
    ALWAYS, LEAVING, NONE;

    companion object {
        fun reduce(a: BorderTraversal?, b: BorderTraversal?): BorderTraversal? {
            return if (a == null) {
                b
            } else if (b == null) {
                a
            } else if (a != ALWAYS && b != ALWAYS) {
                NONE
            } else {
                ALWAYS
            }
        }
    }
}